httpbp/headers.go

```diff
@@ -67,11 +69,21 @@ type EdgeContextHeaders struct {
 // NewEdgeContextHeaders returns a new EdgeContextHeaders object from the given
 // HTTP headers.
 func NewEdgeContextHeaders(h http.Header) EdgeContextHeaders {
+	ec, err := base64.StdEncoding.DecodeString(h.Get(EdgeContextHeader))
+	if err != nil {
```

Comment:
I'm not so sure that we want to throw the error away.

`NewEdgeContextHeaders` is called by `TrustEdgeContext`, and if the decoding failed, `TrustEdgeContext` should also report failure.

Although in most cases returning an empty edge request here will also cause `TrustEdgeContext` to fail, I still prefer this to be done explicitly.